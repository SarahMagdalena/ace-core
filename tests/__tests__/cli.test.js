'use strict';

const ace = require('../runAceCLI');
const pkg = require('../../packages/ace-core/package');

const path = require('path');

describe('Running the CLI', () => {
  test('with no input should fail', () => {
    const { stdout, stderr, status } = ace([]);
    expect(status).toBe(1);
    expect(stderr.trim()).toMatchSnapshot();
    expect(stdout).toMatchSnapshot();
  });
  
  test('with the -h option should print help', () => {
    const { stdout, stderr, status } = ace(['-h']);
    expect(status).toBe(0);
    expect(stderr).toBe('');
    expect(stdout).toMatchSnapshot();
  });
  
  test('with the -v option should print the version number', () => {
    const { stdout, stderr, status } = ace(['-v']);
    expect(status).toBe(0);
    expect(stderr).toBe('');
    expect(stdout.trim()).toBe(pkg.version);
  });
  
  test('with the --version option should print the version number', () => {
    const { stdout, stderr, status } = ace(['--version']);
    expect(status).toBe(0);
    expect(stderr).toBe('');
    expect(stdout.trim()).toBe(pkg.version);
  });
  
  test('on an unexisting document should fail', () => {
    const { stdout, stderr, status } = ace(['unexisting.epub']);
    expect(status).toBe(1);
    expect(stdout.trim()).toMatchSnapshot();
    expect(stderr).toMatchSnapshot();
  });
  
  test('with -o pointing to an existing directory should fail', () => {
    const { stdout, stderr, status } = ace(['-o', 'report', 'foo'], {
      cwd: path.resolve(__dirname, '../data'),
    });
    expect(status).toBe(1);
    expect(stderr).toBe('');
    expect(stdout).toMatchSnapshot();
  });

  test('with --silent and no --outdir should print the JSON report to standard output', () => {
    const { stdout, stderr, status } = ace(['-s', 'base-epub-30'], {
      cwd: path.resolve(__dirname, '../data'),
    });
    expect(status).toBe(0);
    expect(stderr).toBe('');
    expect(() => JSON.parse(stdout)).not.toThrow(SyntaxError);
    const res = JSON.parse(stdout);
    expect(res).toMatchObject({ '@type': 'earl:report' });
  });
});

