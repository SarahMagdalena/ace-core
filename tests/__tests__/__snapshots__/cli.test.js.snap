// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test existing output 1`] = `
"[33mwarn[39m:    Output directory is not empty.

  Running Ace would override the following files or directories:

  - report/report.json
  - report/report.html
  - report/js

  Use option --force to override.

"
`;

exports[`test help 1`] = `
"
  Ace by DAISY, an Accessibility Checker for EPUB

  Usage: ace [options] <input>

  Options:

    -h, --help             output usage information
    -v, --version          output the version number

    -o, --outdir  <path>   save final reports to the specified directory
    -t, --tempdir <path>   specify a custom directory to store the temporary reports
    -f, --force            override any existing output file or directory
        --subdir           output reports to a sub-directory named after the input EPUB

    -V, --verbose          display verbose output
    -s, --silent           do not display any output
    -R, --rules            to get information about all the internal used rules

  Examples
    $ ace -o out ~/Documents/book.epub
"
`;

exports[`test no input 1`] = `"[31merror[39m:   Input required"`;

exports[`test no input 2`] = `
"
  Ace by DAISY, an Accessibility Checker for EPUB

  Usage: ace [options] <input>

  Options:

    -h, --help             output usage information
    -v, --version          output the version number

    -o, --outdir  <path>   save final reports to the specified directory
    -t, --tempdir <path>   specify a custom directory to store the temporary reports
    -f, --force            override any existing output file or directory
        --subdir           output reports to a sub-directory named after the input EPUB

    -V, --verbose          display verbose output
    -s, --silent           do not display any output
    -R, --rules            to get information about all the internal used rules

  Examples
    $ ace -o out ~/Documents/book.epub
"
`;

exports[`test unexisting input 1`] = `
"[32minfo[39m:    Closing logs.
Re-run Ace using the --verbose option to enable full debug logging."
`;

exports[`test unexisting input 2`] = `
"[31merror[39m:   Couldnâ€™t find EPUB file 'unexisting.epub'
"
`;
